src/main/java/com/mraof/minestuck/effects/CreativeShockEffect.java

```diff
@@ -0,0 +1,166 @@
+package com.mraof.minestuck.effects;
+
+import com.mraof.minestuck.Minestuck;
+import com.mraof.minestuck.item.artifact.CruxiteArtifactItem;
+import com.mraof.minestuck.network.MSPacketHandler;
+import com.mraof.minestuck.network.StopCreativeShockEffectPacket;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.item.ChorusFruitItem;
+import net.minecraft.item.EnderPearlItem;
+import net.minecraft.potion.Effect;
+import net.minecraft.potion.EffectType;
+import net.minecraftforge.event.TickEvent;
+import net.minecraftforge.event.entity.player.PlayerEvent;
+import net.minecraftforge.event.entity.player.PlayerInteractEvent;
+import net.minecraftforge.event.world.ExplosionEvent;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+/**
+ * This is an adapted version of Cibernet's code in Minestuck Universe, credit goes to him!
+ * The different levels of amplification increase the range of the effect, allowing the selective limitation of both creative and non creative players.
+ * 0 = cant directly cause block breakage/placement outside of creative, 1 = cant access redstone machinery gui outside of creative, 2 = cant use mobility items outside of creative,
```

Comment:
I'm wondering if you should make redstone machinery gui access uniform to redstone repeaters, redstone dust and comparators, which require `player.abilities.mayBuild` to be true for them to be "configured" with right-click.

Reply:
If this was the case, would you also be suggesting the removal of that effect strength tier?

Reply:
Since this effect wont be properly utilized until Tiered Dungeons, I think I will keep this the same for now and continue looking into reworks of the effect once it becomes relevant